### ISSUE TITLE
Adding credits / roles / attribution to records

### ISSUE THREAD

**Issue by joeytakeda** [sfu-dhil/digital-mary#75]
### Describe the feature

Add the ability to note author(s), researcher(s), and editor(s) to individual records, which should also be reflected in the citation.

### How will this feature enhance your project and further the project’s overall goals? Who will benefit from this feature (i.e. all users; the project team)?

There are various ways that entries are added to the database: in some cases, an RA may research, write, and create the entry; in others, a student or researcher without write permissions to the database may do the research, author the entry, or edit the record. However, there is no way currently to specify the roles and responsibilities of individuals as the list of "Contributions" is generated from whoever has last altered a record.

### Are there any alternatives for this feature? Are there any consequences for your project if this feature is not implemented?

N/A

### Please share any examples of this feature, or one similar to it, that you have encountered in other contexts.

N/A

### Additional information

(From DM mtg 2023-12-11)

**Comment by joeytakeda** [comment]
Per discussion, there are a few ways that this could be approached:

1. Separate fields for author, editor, et cetera, which would have a display name (or perhaps Surname, Forename) in order to capture particular roles. This is the most straightforward mechanism, and would provide a simple way for providing information. Development with this approach would be much faster and could be expanded upon later, if necessary. 

2. Create a many-to-many relationship between Roles, Contributors, and Contributions—something akin to:

```mermaid 
erDiagram

    Role {
      int roleID PK 
      string roleName
      string roleDescription
      string roleURI "Standardized URI, probably from CRedit or LOC"
    }

    Contributor {
        int contributorID
        string forename
        string surname
        string displayName
        string biographicalNote 
    }

   Contribution 1--1 Role: has
   Contribution 1--1 Contributor: has

   Contribution {
     int itemID
     int contributorID
     int roleID
   }
```

While a bit more complicated, I think this would be a bit more flexible in future:

* using a standardized role taxonomy (e.g. from [CRedit](https://credit.niso.org/) or [LOC](https://www.loc.gov/marc/relators/relaterm.html)) could be more readily converted into a bibliographic standards (i.e. BIBTex), which would also allow for more sophisticated automated citations (using [`citeproc`](https://github.com/juris-m/citeproc-js) or similar tools)
* while individuals may not necessarily need biographies, future contributors may need them; plus, anecdotally, specifying aggregating contributions can be useful for a variety of purposes (e.g. generating a page of all contributors, for instance, or creating pages for each contributor, which they can point back to in e.g. their CVs)
* this would allow ensure that _attribution_ remains separated from _revision_: at the moment, revision tracking is doing double duty, which can be misleading

**Comment by andrew-gardener** [comment]
closed by 09d82dd8d4f84fccbbd52796068a326f6e1d2d61
